### PR TITLE
PIL-245 Add services tab

### PR DESCRIPTION
**Pull request by graszka22**
About changes in the Exchange screen:
I've moved the initialization of the selection of assets from `componentDidUpdate` to `componentDidMount` because it was causing infinite loop of re-renders. I think it's not necessary anyway to check the navigation params every time the component updates since they're not changed anywhere now - they're just passed from the previous screen and that will be handled in `componentDidMount`.

### CODE REVIEW

File: src/screens/Services/Services.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import type { RootReducerState } from 'reducers/rootReducer';
+
+type Props = {
+    theme: Theme,
+    offers: Offer[],
+    navigation: NavigationScreenProp<*>,
+};
+
+class ServicesScreen extends React.Component<Props> {
+    getServices = () => {
```
Comment (jsidorenko) [inline src/screens/Services/Services.js]: Pls check your IDE settings, we're using 2 spaces as default indentation

File: src/screens/Services/package.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "name": "Services.js",
```
Comment (jsidorenko) [inline src/screens/Services/package.json]: Same as above

File: src/screens/Services/Services.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            keyExtractor={(item) => item.key}
+            renderItem={this.renderServicesItem}
+            contentContainerStyle={{ width: '100%', padding: 20, paddingBottom: 40 }}
+          />
+        </ContainerWithHeader>
+      );
+    }
+}
+
+const mapStateToProps = ({ exchange: { data: { offers } } }: RootReducerState): $Shape<Props> => ({
```
Comment (jsidorenko) [inline src/screens/Services/Services.js]:
```suggestion
const mapStateToProps = ({
  exchange: { data: { offers } },
}: RootReducerState): $Shape<Props> => ({
```

File: src/screens/Services/Services.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  offers,
+});
+
+export default withTheme(connect(mapStateToProps)(ServicesScreen));
```
Comment (jsidorenko) [inline src/screens/Services/Services.js]: Missed a new line at the end of file

File: src/screens/Services/Services.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+import type { NavigationScreenProp } from 'react-navigation';
+import Intercom from 'react-native-intercom';
+import { ListCard } from 'components/ListItem/ListCard';
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import { EXCHANGE } from 'constants/navigationConstants';
+import { USD, ETH } from 'constants/assetsConstants';
+import { getThemeColors } from 'utils/themes';
+import type { Theme } from 'models/Theme';
+import type { Offer } from 'models/Offer';
+import { withTheme } from 'styled-components/native';
```
Comment (jsidorenko) [inline src/screens/Services/Services.js]: @vyvil here and in Assets we're importing `withTheme` from 'styled-components/native' while on other screens importing it from 'styled-components' - which route is correct?

Comment (vyvil) [reply src/screens/Services/Services.js]: There is no big difference - it's the same withTheme being exported from 'styled-components' and 'styled-components/native';

For a sake of consistency we could stick to 'styled-components/native' (as we get our configured `styled` from there as well.

**Comment by vyvil** [comment]
As it is a quick one, I'll push a small addition that is kind of related to the changes - it will fix an navigation issue:
If user navigates to exchange from asset screen and then tries to get back, he/she is taken to Services tab rather than asset screen.

**Comment by graszka22** [comment]
I've updated the code, fixing code style. As a side note, shouldn't such things be listed in the `.eslint.rc`? That way it'll be formatted and checked automatically. I tried to add it, but there are a lot of files that don't have correct indentation and I'd update half of the project 😜

**Comment by jsidorenko** [comment]
We have the '.editorconfig' file for this

File: src/screens/Services/package.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "Services.js",
+  "version": "0.0.0",
+  "private": true,
+  "main": "./Services.js"
+}
```
Comment (grilix) [inline src/screens/Services/package.json]: This one is also missing a newline at the end.

File: src/screens/Services/Services.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+        title: 'Peer-to-peer trading',
+        body: 'Swap tokens directly with others. safe, secure, anonymous',
+        disabled: true,
+        label: 'soon',
+      },
+      {
+        key: 'buyCryptoWithFiat',
+        title: 'Buy crypto with fiat',
+        body: 'USD, GBP, EUR supported',
+        action: () => navigation.navigate(EXCHANGE, { fromAssetCode: USD, toAssetCode: ETH }),
```
Comment (vyvil) [inline src/screens/Services/Services.js]: `fromAssetCode` could be default currency - it is either baseFiatCurrency (if user has set it), that is stored in appSettings, or defaultFiatCurrency (which is a constant set in assetsConstants.js).

File: src/screens/Services/Services.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+          rightItems: [{ link: 'Support', onPress: () => Intercom.displayMessenger() }],
+          leftItems: [{ title: 'Services' }],
+        }}
+        inset={{ bottom: 'never' }}
+      >
+        <FlatList
+          data={services}
+          keyExtractor={(item) => item.key}
+          renderItem={this.renderServicesItem}
+          contentContainerStyle={{ width: '100%', padding: 20, paddingBottom: 40 }}
```
Comment (vyvil) [inline src/screens/Services/Services.js]: I believe, horizontal paddings should be 16px - `spacing.layoutSides`

File: src/components/LabelBadge/LabelBadge.js [hunk 9]
```diff
@@ -26,10 +26,11 @@ type Props = {
   label: string,
   containerStyle?: Object,
   labelStyle?: Object,
+  color?: string,
 }
 
 const BadgeWrapper = styled.View`
-  background-color: ${themedColors.positive};
+  background-color: ${props => props.color || themedColors.positive};
```
Comment (poocart) [inline src/components/LabelBadge/LabelBadge.js]:
```suggestion
  background-color: ${({ color }) => color || themedColors.positive};
```

File: src/components/ListItem/ListCard.js [hunk 10]
```diff
@@ -120,6 +126,12 @@ export const ListCard = (props: Props) => {
             <TitleWrapper>
               <CardTitle style={titleStyle}>{title}</CardTitle>
               {!!label && <Label>{label}</Label>}
+              {!!labelBadge && (
+                <LabelBadge
+                  label={labelBadge.label}
+                  labelStyle={{ fontSize: fontSizes.regular }}
+                  color={labelBadge.color}
+                />)}
```
Comment (poocart) [inline src/components/ListItem/ListCard.js]:
```suggestion
                />
              )}
```

**Pull request merged**